src/Controller/RestaurantController.php

```diff
@@ -229,16 +262,30 @@ public function listAction(Request $request,
             $matches = $repository->findByLatLng($latitude, $longitude);
         } else {
 
-            $cacheKey = sprintf('restaurant.list.ids|%s', LocalBusiness::getKeyForType($type));
+            $cacheKey = $this->getShopsListCacheKey($request, $type);
 
-            $restaurantsIds = $projectCache->get($cacheKey, function (ItemInterface $item) use ($repository, $type) {
+            // for filtering we need in query param the full type instead of the key
+            $request->query->set('type', $type);
+
+            if ($request->query->has('cuisine')) {
```

Comment:
Instead of using comma-separated values in the params, we should make usage of native PHP management of arrays. The URLs would look like `?cuisine[]=indian&cuisine[]=asian`. Then `$request->query->get('cuisine')` would directly return an array.

Reply:
I've changed to use arrays instead of comma-separated string ([commit](https://github.com/coopcycle/coopcycle-web/pull/3176/commits/0da0eff26896155ae397cd1dc18e4fa4170084ee)) 
Something that I can't figure out how to change is that the array has the index for each value and the URL looks like this:
`/shops?cuisine[0]=asian&cuisine[1]=indian`